samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -262,3 +262,23 @@ def expect_minimum_argument_count!(arguments, min_count)
   false
 }
 
+ZIL_BUILTINS[:OBJECT] = define_for_evaled_arguments { |arguments, context|
+  # Objects have a name, and a list of properties and values
+  expect_minimum_argument_count!(arguments, 1)
+  
+  object_name, *object_properties = arguments
+
+  object = { properties: {} }
+  object[:name] = object_name
```

Comment:
I *think* that the object knowing it's own name isn't required, I may be wrong though